pkg/csi_driver/controller.go

```diff
@@ -183,7 +182,7 @@ func (s *controllerServer) CreateVolume(ctx context.Context, req *csi.CreateVolu
 
 		if err != nil {
 			klog.Errorf("CreateVolume returned an error %v, for request %+v", err, req)
-			return nil, err
+			return nil, file.StatusError(err)
```

Comment:
we still need to return unavailable code to the caller of `CreateVolume`, now since we are at outermost, we need to unwrap the temporary error with the unavailable code.

see https://github.com/kubernetes-sigs/gcp-compute-persistent-disk-csi-driver/pull/1664 for detail. follow how the temporary error returned from the "internal" methods are propagated and see how it is unwrapped at the return point of `CreateVolume` call.

Reply:
the `file.StatusError` does the unwrapping of the error. I added a new check for temporary error [here](https://github.com/kubernetes-sigs/gcp-filestore-csi-driver/blob/60c6e085ff3bfa0682d420b314a3ce435b57fc98/pkg/cloud_provider/file/file.go#L718) that handles it. 

I think what is missing from this PR is the handling the metrics for new Temporary error types. This will be done in separate PR.

Reply:
@leiyiz  WDYT?